---
- hosts: all
  ignore_unreachable: True
  gather_facts: False
  tasks:
    - name: Testing "Ansible connectivity"
      ping:
       data: "ok, Ansible ready!"
      register: result
      tags: ansible, check
    - meta: clear_host_errors
    - debug:
        var: result
        verbosity: 1
    - name: Timeout hook
      set_fact:
        host_error: "connection timed out"
      when: "'Connection timed out' in result.msg|d('')"
      no_log: true
    - name: NACK hook
      set_fact:
        host_error: "connection refused"
      when: "'Connection refused' in result.msg|d('')"
      no_log: true
    - name: Handling connection errors
      fail:
        msg: "Host connection failed with error {{ host_error|upper }} details: {{ result.msg|d('') }}"
      when: "host_error is defined"

- name: Deploy Ansible provisioning environment
  import_playbook: deploy_ssh.yml
  when: result is defined and (result.ping is not defined or (provision_force is defined and provision_force))
  ignore_errors: True
  tags: deploy, ssh
